web/packages/teleterm/src/ui/ConnectMyComputer/DocumentConnectMyComputerSetup/DocumentConnectMyComputerSetup.tsx

```diff
@@ -82,9 +84,32 @@ function Information(props: { onSetUpAgentClick(): void }) {
 
 function AgentSetup() {
   const ctx = useAppContext();
+  const { rootClusterUri } = useWorkspaceContext();
 
   const [setUpRolesAttempt, runSetUpRolesAttempt] = useAsync(
-    useCallback(() => wait(1_000), [])
+    useCallback(async () => {
```

Comment:
I think we should add `retryWithRelogin` somewhere. 
I would probably do this in the `for` loop in `runSteps`, so it would handle join token creation as well.

Reply:
I added `useRetryWithRelogin`, let me know what you think.

I think we've never had to write something like `useRetryWithRelogin` because we didn't care much about proper `useCallback` deps in any place that used `retryWithRelogin`. ;f

Reply:
Maybe we could re-throw the error?

```ts
    for (const action of actions) {
      try {
        await retryWithRelogin(ctx, rootClusterUri, async () => {
          const [, error] = await action();
          if (error) {
            throw error;
          }
        });
      } catch (error) {
        break;
      }
    }
```

Reply:
I'm not sure if that's better than `useRetryWithRelogin`, especially since by using `useRetryWithRelogin` we signal explicitly which actions need to account for relogin and which don't.